go.mod

```diff
@@ -74,17 +74,19 @@ require (
 	k8s.io/kube-aggregator v0.29.7
 	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
 	k8s.io/kube-proxy v0.29.7
+	k8s.io/kube-state-metrics/v2 v2.11.0
 	k8s.io/kubelet v0.29.7
 	k8s.io/metrics v0.29.7
 	k8s.io/pod-security-admission v0.29.7
 	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
 	sigs.k8s.io/controller-runtime v0.17.5
-	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20231015215740-bf15e44028f9 // v0.16.3
 	sigs.k8s.io/controller-tools v0.14.0
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1
 	sigs.k8s.io/yaml v1.4.0
 )
 
+require sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20231015215740-bf15e44028f9 // v0.16.3
```

Comment:
I tried to capture the reasons in commit [`d281b7b`](https://github.com/gardener/gardener/pull/9941/commits/d281b7ba2f0a7efad3065c98d464b13126d3b92b). 

But in short, when this line is inside the `require` block from above, go is not able to understand that such block is for the direct dependencies and that new direct dependencies should be added there as well. So, when I added the KSM dependency via `go get` and `go mod tidy`,  go created a new `require` block only for the KSM. @istvanballok and I investigated this and found the culprit is this dependency because it has the version commented at the end (`... // v0.16.3`). After extracting this dependency in a new `require` block, `go` correctly added the new KSM dependency in the right block.

But this was just a proposal. We can keep it in one block altogether. I will address this 👍🏻

Reply:
Unfortunately it does not :(

Reply:
OK. I can live with both, either put it back, or keep it in the other block, but then maybe add a short comment why this is (to prevent we move this back up later)

Reply:
I put it back. I think it's going to be more confusing than helpful :)